docs/guides/index.md

```diff
@@ -9,58 +9,17 @@ and features.
 :hidden:
 
 *
-esp32/README
-nxp/README
-ti/ti_matter_overview
 ```
 
 ## Build Guides
 
 -   [Building](./BUILDING.md)
```

Comment:
Should we move this to Getting Started?

In general, this "Guides" grouping doesn't seem that useful now with most of the other content moved out of it.  I would maybe just remove it and put the content in the other groupings as needed.

Reply:
yeah, the eventual goal is to get rid of the "Guides" directory, but this PR is JUST moving the platform stuff out. I want to be able to have the platform folks take a look and make sure they're OK with this without adding in a bunch of other changes that don't affect them.

Reply:
discussed offline - handling in follow up